### PR TITLE
Remove guarding check on computed field with field_serializer

### PR DESCRIPTION
**Pull request by nix010**
## Change Summary

Remove guarding check on computed field with field_serializer

## Related issue number

#9683 
## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @sydney-runkle

### CODE REVIEW

**Comment by nix010** [comment]
please review

File: tests/test_serialize.py [hunk 1]
```diff
@@ -1073,22 +1072,21 @@ def quadruple_x_plus_one(self) -> Annotated[int, PlainSerializer(lambda v: v + 1
...
+        @property
+        def two_x(self) -> int:
+            return self.x * 2
 
-            @computed_field
-            @property
-            def two_x(self) -> int:
-                return self.x * 2
+        @field_serializer('two_x')
+        def ser_two_x_bad_signature(self, v, _info):
```
Comment (sydney-runkle) [inline tests/test_serialize.py:1085]: Should probably rename this - doesn't have a bad signature now.

Comment (sydney-runkle) [reply tests/test_serialize.py:1085]: Can probably delete this test honestly - not sure of the value here anymore, especially given the other test added above.

**Review by sydney-runkle — CHANGES_REQUESTED** [review]
A few change requests, thanks!

**Comment by sydney-runkle** [comment]
LGTM overall - happy to merge after we fix up the tests a bit.

Confirming with the team that there was no reason not to support this, before we merge as well.

**Comment by sydney-runkle** [comment]
See https://github.com/pydantic/pydantic/pull/6965 where this was added - there's some other relevant code that we should update - docstrings, etc.

**Comment by nix010** [comment]
@sydney-runkle removed. please review again.

> See #6965 where this was added - there's some other relevant code that we should update. We can remove the `computed_field: bool` from the `_apply_field_serializers` function, as well :).

File: pydantic/_internal/_generate_schema.py [hunk 2]
```diff
@@ -1972,7 +1972,6 @@ def _computed_field_schema(
         return_type_schema = self._apply_field_serializers(
             return_type_schema,
             filter_field_decorator_info_by_field(field_serializers.values(), d.cls_var_name),
-            computed_field=True,
         )
```
Comment (sydney-runkle) [inline pydantic/_internal/_generate_schema.py:1975]: I believe you can remove the `computed_field` argument from the `_apply_field_serializers` function as well.

https://github.com/pydantic/pydantic/blob/a6dc87285f93f90c2d5c298ee7c52f5d7e878194/pydantic/_internal/_generate_schema.py#L2196C9-L2201

Comment (sydney-runkle) [reply pydantic/_internal/_generate_schema.py:1975]: Otherwise, lgtm! Thanks!

Comment (nix010) [reply pydantic/_internal/_generate_schema.py:1975]: https://github.com/pydantic/pydantic/pull/10390/files#diff-2ebfa892ec1d09440eefd4b22f488ec38c1cde365394e5a4cea6edb484827ba5R2198-R2199
@sydney-runkle I've already removed it in this commit.

Comment (sydney-runkle) [reply pydantic/_internal/_generate_schema.py:1975]: Oh great! Not sure how I missed that 🤨 thanks!

**Review by sydney-runkle — APPROVED** [review]
Great work! Thanks so much @nix010!

**Pull request merged**